rpcserver.go

```diff
@@ -7246,19 +7246,32 @@ func (r *rpcServer) EncodeCustomRecords(_ context.Context,
 func (r *rpcServer) SendPayment(req *tchrpc.SendPaymentRequest,
 	stream tchrpc.TaprootAssetChannels_SendPaymentServer) error {
 
+	if len(req.AssetId) > 0 && len(req.GroupKey) > 0 {
+		return fmt.Errorf("cannot set both asset id and group key")
+	}
+
 	if req.PaymentRequest == nil {
 		return fmt.Errorf("payment request must be specified")
 	}
 	pReq := req.PaymentRequest
 	ctx := stream.Context()
 
-	// Do some preliminary checks on the asset ID and make sure we have any
-	// balance for that asset.
-	if len(req.AssetId) != sha256.Size {
-		return fmt.Errorf("asset ID must be 32 bytes")
+	var specifier asset.Specifier
+
+	assetID, groupKey, err := parseAssetSpecifier(
+		req.AssetId, "", req.GroupKey, "",
+	)
+	if err != nil {
+		return err
+	}
+
+	switch {
```

Comment:
Should we add this `switch` and the error check above as a new function `asset.NewExclusiveSpecifier()`?
To make sure we use the same semantics everywhere.